src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -184,12 +184,17 @@ private void validateTable(TableModel table, Errors errors, SchemaValidationCont
     String tableName = table.getName();
     List<ColumnModel> columns = table.getColumns();
     List<String> primaryKeyList = table.getPrimaryKey();
+    List<String> columnNames = new ArrayList<>();
+    if (columns == null || columns.isEmpty()) {
+      errors.rejectValue(
+          "schema", "IncompleteSchemaDefinition", "Each table must contain at least one column");
+    } else {
+      columnNames.addAll(columns.stream().map(ColumnModel::getName).collect(Collectors.toList()));
+    }
```

Comment:
Instead of
```
      columnNames.addAll(columns.stream().map(ColumnModel::getName).collect(Collectors.toList()));
 ```
I would write
```
      columns.stream().map(ColumnModel::getName).forEach(columnNames::add);
```
Since that avoids creating an intermediate list result. But they both do the same thing so it's up to you...

Also it seems crazy to me that we have to check `table.getColumns() != null` given that it has a `@NonNull` validation constraint. But I guess that's how a custom spring validation works?

Reply:
@pshapiro4broad  - You're right, we don't need those null checks. Should be fixed.